src/client.ts

```diff
@@ -24,46 +39,27 @@ export function makeClient(config: LanguageConfig): vscode.Disposable {
     workspaceFolder: vscode.workspace.workspaceFolders?.[0]
   }
 
-  return utils.bracket<child_process.ChildProcessWithoutNullStreams, vscode.Disposable>({
-    open: () => child_process.spawn(serverCommand, [], { env: process.env }),
-    close: server => server.kill(),
-    use: server => {
-      return utils.bracket<lsp.LanguageClient, vscode.Disposable>({
-        open: () => {
-          let client = new lsp.LanguageClient(
-            clientId,
-            `${alloglot.root} language client for ${languageId}`,
-            () => Promise.resolve(server),
-            clientOptions,
-            false
-          )
-          client.start()
-          return client
-        },
-        close: client => client.stop(),
-        use: client => {
-          return vscode.Disposable.from({
-            dispose: () => {
-              client.stop()
-              server.kill()
-              output.dispose()
-            }
-          })
-        }
-      })
-    }
-  })
-}
+  output.append(`${alloglot.root} debug: Starting language client...\n`)
 
-namespace utils {
-  export function bracket<T, U>(args: { open: () => T, close: (t: T) => void, use: (t: T) => U }): U {
-    let t: T | undefined = undefined
-    try {
-      t = args.open()
-      return args.use(t)
-    }
-    finally {
-      if (t) args.close(t)
-    }
```

Comment:
so it turns out i was closing the server just as soon as i was opening it, i think. actually, i'm not sure what the problem was.